src/directory/managed_directory.rs

```diff
@@ -207,26 +210,99 @@ impl ManagedDirectory {
         }
         Ok(())
     }
+
+    /// Verify checksum of a managed file
+    pub fn validate_checksum(&self, path: &Path) -> result::Result<bool, OpenReadError> {
+        let reader = self.directory.open_read(path)?;
+        let data = reader.as_slice();
+        let len = data.len() - CRC_SIZE;
+        let mut hasher = Hasher::new();
+        hasher.update(&data[..len]);
+        let crc = hasher.finalize().to_be_bytes();
+        Ok(data[len..]==crc)
+    }
+
+    /// List files for which checksum does not match content
+    pub fn list_damaged(&self) -> result::Result<HashSet<PathBuf>, OpenReadError> {
+        let mut hashset = HashSet::new();
+        let meta_informations_rlock = self
+            .meta_informations
+            .read()
+            .expect("Managed directory rlock poisoned in list damaged.");
```

Comment:
I'm ok with unwrapping rlock... under certain conditions.

So the key idea is that you want to make sure that the lock cannot be poisoned... AND make it trivial 
for the reader that it cannot be poisoned. 

Here the conditions are not met because you hold the lock, while you perform a lot of IO, some of them 
depending on user defined code (if there is a custom directory involved for instance).

Can you clone `managed_paths` right after acquiring the lock and release it right away?

Reply:
I did made sure to not panic myself, but did not consider user code being run here. I'll do the clone